### PR TITLE
Fix start_auth_session resource leak

### PR DESCRIPTION
**Pull request by Zha0Chan**
The session handle returned by _start_auth_session_. If there is an error in subsequent function calls, it is best to release it immediately.

### CODE REVIEW

**Review by ionut-arm — DISMISSED** [review]
Thanks for the patch, it makes sense to me, I guess we haven't noticed because those commands don't fail that often.

**Review by Superhepper — DISMISSED** [review]
Nice catch

File: tss-esapi/src/abstraction/ak.rs [hunk 1]
```diff
@@ -182,11 +187,16 @@ pub fn create_ak<IKC: IntoKeyCustomization>(
...
-        session_attributes_mask,
-    )?;
+    context
+        .tr_sess_set_attributes(
+            policy_auth_session,
+            session_attributes,
+            session_attributes_mask,
+        )
+        .or_else(|e| {
+            ctx.flush_context(SessionHandle::from(policy_auth_session).into())?;
```
Comment (Superhepper) [inline tss-esapi/src/abstraction/ak.rs:197]: There is no `ctx` in this scope.

File: tss-esapi/src/context.rs [hunk 2]
```diff
@@ -305,7 +305,11 @@ impl Context {
             .with_decrypt(true)
             .with_encrypt(true)
             .build();
-        self.tr_sess_set_attributes(auth_session, session_attributes, session_attributes_mask)?;
+        self.tr_sess_set_attributes(auth_session, session_attributes, session_attributes_mask)
+            .or_else(|e| {
+                ctx.flush_context(SessionHandle::from(auth_session).into())?;
```
Comment (Superhepper) [inline tss-esapi/src/context.rs:310]: there is no `ctx` in this scope either.

**Review by Zha0Chan — COMMENTED** [review]
Fixed some typos and compiled it successfully

**Comment by Superhepper** [comment]
Have you signed the commit(s)? I think I saw the DCO check failed.

**Review by Superhepper — APPROVED** [review]

**Review by ionut-arm — APPROVED** [review]

**Pull request merged**